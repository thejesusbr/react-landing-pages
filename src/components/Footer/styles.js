import styled, { css } from 'styled-components';
import { SimpleText as TextComponent } from '../TextComponent/styles';
import { MyStyle as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    background: ${theme.colors.lightColor};
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.secondaryColor};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
