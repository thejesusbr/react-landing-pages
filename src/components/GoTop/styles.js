import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.lightColor};
    border: 1px solid ${theme.colors.lightColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${theme.font.sizes.xlarge};
    height: ${theme.font.sizes.xlarge};
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;

    & svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
