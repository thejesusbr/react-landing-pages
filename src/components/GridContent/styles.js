import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 80%;
    text-align: center;
    margin: auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
