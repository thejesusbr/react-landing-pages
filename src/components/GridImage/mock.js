export default {
  background: false,
  sectionId: 'intro',
  title: 'Image Grid',
  description:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Nobis nam nisi ad commodi sit velit mollitia tenetur non dolorem ut dolor nulla alias, reprehenderit nostrum? Nihil maxime assumenda saepe necessitatibus!',
  grid: [
    {
      altText: 'Texto alternativo 1',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Texto alternativo 2',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Texto alternativo 3',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Texto alternativo 4',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Texto alternativo 5',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Texto alternativo 6',
      imgSrc: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
};
