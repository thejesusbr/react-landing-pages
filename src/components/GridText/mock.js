export default {
  background: false,
  sectionId: 'grid',
  title: 'Grid Section',
  description: 'Lorem ipsum dolor sit, amet consectetur adpiscing elit.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Beatae totam voluptatibus ex asperiores. Accusantium ipsam debitis molestias commodi quis temporibus quod eos cum quibusdam est perferendis pariatur veritatis, dolorum in.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Beatae totam voluptatibus ex asperiores. Accusantium ipsam debitis molestias commodi quis temporibus quod eos cum quibusdam est perferendis pariatur veritatis, dolorum in.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Beatae totam voluptatibus ex asperiores. Accusantium ipsam debitis molestias commodi quis temporibus quod eos cum quibusdam est perferendis pariatur veritatis, dolorum in.',
    },
  ],
};
