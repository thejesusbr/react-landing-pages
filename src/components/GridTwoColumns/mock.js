export default {
  component: 'section.section-two-columns',
  sectionId: 'intro',
  title: 'Home',
  text: 'Já imaginou produzir sua própria energia e reduzir…ue agride menos o meio-ambiente? Agora você pode!',
  imgSrc:
    'https://res.cloudinary.com/singularity-softworks/i…d/v1647653269/piqsels_com_id_sbrnx_c81d065c8a.jpg',
  background: true,
};
