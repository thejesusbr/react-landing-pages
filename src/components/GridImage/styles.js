import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { SimpleText as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    height: auto;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
