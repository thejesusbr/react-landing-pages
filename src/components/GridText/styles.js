import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { SimpleText as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.huge};
      top: -1rem;
      left: -${theme.spacings.xlarge};
      transform: rotate(5deg);
    }

    @media ${theme.media.lteMedium} {
      ${HeadingContainer}::before {
        font-size: ${theme.font.sizes.xlarge};
        top: -0.75rem;
        left: -${theme.spacings.large};
      }
    }
  `}
`;
