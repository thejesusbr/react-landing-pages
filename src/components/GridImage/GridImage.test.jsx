import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { GridImage } from '.';
import mock from './mock';

describe('<GridImage />', () => {
  it('should render with background', () => {
    renderTheme(<GridImage {...mock} background={true} />);
    expect(
      screen.getByRole('heading', { name: 'Image Grid' }),
    ).toBeInTheDocument();
  });

  it('should render when no sectionId is provided', () => {
    renderTheme(
      <GridImage {...mock} sectionId={undefined} background={true} />,
    );
    expect(
      screen.getByRole('heading', { name: 'Image Grid' }),
    ).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = renderTheme(
      <GridImage {...mock} background={undefined} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
