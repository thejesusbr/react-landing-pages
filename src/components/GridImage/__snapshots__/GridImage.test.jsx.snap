// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should match the snapshot 1`] = `
.c3 {
  color: #0a1128;
  font-size: 5.6rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 6.4rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  max-width: 100%;
  height: auto;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c1 {
  max-width: 120rem;
  margin: auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #f3f3f3;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div
  class="c0"
  id="intro"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        Image Grid
      </h2>
      <div
        class="c4 c5"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nobis nam nisi ad commodi sit velit mollitia tenetur non dolorem ut dolor nulla alias, reprehenderit nostrum? Nihil maxime assumenda saepe necessitatibus!
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <img
            alt="Texto alternativo 1"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=1"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Texto alternativo 2"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=2"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Texto alternativo 3"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=3"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Texto alternativo 4"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=4"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Texto alternativo 5"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=5"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Texto alternativo 6"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=6"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
